sirepo/tornado.py

```diff
@@ -6,8 +6,30 @@
 """
 from pykern.pkdebug import pkdlog, pkdexc
 import sirepo.util
+import tornado.locks
 import tornado.queues
 
+class Event(tornado.locks.Event):
+    class _OrderedSet:
+
+        def __init__(self):
+            self.vals = list()
+
+        def __iter__(self):
+            return iter(self.vals)
+
+        def add(self, val):
+            if val in self.vals:
+                return
+            self.vals.append(val)
+
+        def remove(self, val):
+            self.vals.remove(val)
+
+    def __init__(self):
```

Comment:
This doesn't require knowledge of _value or its initial value.

```py
def __init__(self):
     super().__init__()
     self._waiters = self._Waiters()
```

Reply:
https://github.com/radiasoft/sirepo/pull/2731/commits/6960d7d6b3672704a51babc454d7c6b07a162c87